File: frontend-react/src/types/CVGenerationRequestDto.ts
import { KeyValueObject } from '@/types/KeyValueObject.ts'

export interface CVGenerationRequestDto {
  includedWorkExperience: undefined | IncludedCVItem[]
  includedEducation: undefined | IncludedCVItem[]
  includedProjects: undefined | IncludedCVItem[]
  includedSkills: undefined | IncludedCVItem[]
  templateOptions: undefined | KeyValueObject<string>
}

export interface IncludedCVItem {
  id: number,
  includeDescription: boolean
}

File: backend/src/main/kotlin/ch/streckeisen/mycv/backend/cv/generator/CVGenerationRequestDto.kt
package ch.streckeisen.mycv.backend.cv.generator

data class CVGenerationRequestDto(
    val includedWorkExperience: List<IncludedCVItem>?,
    val includedEducation: List<IncludedCVItem>?,
    val includedProjects: List<IncludedCVItem>?,
    val includedSkills: List<IncludedCVItem>?,
    val templateOptions: Map<String, String>?
)

data class IncludedCVItem(
    val id: Long,
    val includeDescription: Boolean = true
)